Update Collector configuration telemetry section (#3483)
open-telemetry/opentelemetry.io: content/en/docs/collector/configuration.md

### Telemetry

Telemetry is where the telemetry for the Collector itself can be configured. It
has two subsections: `logs` and `metrics`.

The `logs` subsection allows configuration of the logs generated by the
Collector.

By default, the Collector will write its logs to `stderr` with a log level of
`INFO`. You can also add static key-value pairs to all logs using the
`initial_fields` section.

View the full list of the `logs` configuration options
[here](https://github.com/open-telemetry/opentelemetry-collector/blob/7666eb04c30e5cfd750db9969fe507562598f0ae/config/service.go#L41-L97).

The `metrics` subsection allows configuration of the metrics generated by the
Collector.

> By default, the Collector will generate basic metrics about itself and its
> components, and expose them on port `8888` and path `/metrics`, in Prometheus
> format, so that we can scrape them at `http://localhost:8888/metrics`.

Reviewer: ```suggestion
By default, the Collector generates basic metrics about itself and its
components in Prometheus format; you can scrape them at <http://localhost:8888/metrics>.
```

Author: Originally, the doc mentions that the port `8888` is exposed locally but this is NOT true after checking out the code [here](https://github.com/open-telemetry/opentelemetry-collector/blob/main/otelcol/unmarshaler.go#L63). So I'd suggest to make it clearer that the metrics are exposed at port `8888` for all interfaces by default (which I'm still trying to confirm whether it's aligned with the design in issue [#28861](https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/28861)) but can be configured in the `telemetry` section.

So I'd assume that the code is correct and may enhance this further if it's agreeable with something like this:
```
By default, the Collector generates basic metrics about itself and its
components, and exposes them in Prometheus format at `:8888/metrics`
of all network interfaces, which can be further controlled in `telemetry`'s `metrics`
subsection; you can scrape them at `http://localhost:8888/metrics`.
```

Reviewer 2: @jpkrohling @brightzheng100 can this conversation be resolved?

> View the full list of the `metrics` configuration options
> [here](https://github.com/open-telemetry/opentelemetry-collector/blob/7666eb04c30e5cfd750db9969fe507562598f0ae/config/service.go#L99-L111).

Reviewer: I know that this was part of the text before, but I'd suggest dropping this link into a repo source file pinned at a specific commit. That is brittle. Besides, there are only two config options, both of which are illustrated in this page.

```suggestion
```

Author: Do you think it's a good idea to exact these options out for both `logs` and `metrics` sections from the code and make them part of the doc with the necessary explanation?

Reviewer 2: @jpkrohling @brightzheng100 can this conversation be resolved?

### Telemetry

The `telemetry` is where the telemetry for the Collector itself can be
configured. Collector's own telemetry can be useful when troubleshooting
Collector issues. It consists of two subsections: `logs` and `metrics`.

The `logs` subsection lets you configure how the logs can be generated by the
Collector. By default, the Collector writes its logs to `stderr` with a log
level of `INFO`. You can also add static key-value pairs to all log entries with
the `initial_fields` to enrich the logging context. As per the `LogsConfig`
defined in v{{%
param vers %}} [here](https://github.com/open-telemetry/opentelemetry-collector/blob/v{{%
param vers %}}/service/telemetry/config.go), the `logs` configuration options are:

- `level`: sets the minimum enabled logging level, default `INFO`.
- `development`: puts the logger in development mode, default `false`.
- `encoding`: sets the logger's encoding, default `console`. Example values are

>   "json", "console".

Reviewer 2: ~~~suggestion
  `json`, `console`.
~~~

      address: 0.0.0.0:8888
```

> Please note that it's possible to scrape the Collector's metrics by using a
> [`Prometheus` receiver](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/prometheusreceiver)

Reviewer: ```suggestion
Note that it's possible to scrape the Collector's metrics by using a
[Prometheus receiver](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/prometheusreceiver)
```
